internal/common/validation/podspec.go

```diff
@@ -33,14 +33,36 @@ func ValidatePodSpec(spec *v1.PodSpec, maxAllowedSize uint) error {
 		if len(container.Resources.Requests) == 0 {
 			return fmt.Errorf("container %v has no resource requests specified", container.Name)
 		}
-
+		err = validateContainerResource(container.Resources.Limits, minJobResources, container.Name, "limit")
+		if err != nil {
+			return err
+		}
+		err = validateContainerResource(container.Resources.Requests, minJobResources, container.Name, "request")
+		if err != nil {
+			return err
+		}
 		if !resourceListEquals(container.Resources.Requests, container.Resources.Limits) {
 			return fmt.Errorf("container %v does not have resource request and limit equal (this is currently not supported)", container.Name)
 		}
 	}
 	return validatePorts(spec)
 }
 
+func validateContainerResource(
+	resourceSpec v1.ResourceList,
+	minJobResources v1.ResourceList,
+	container_name string,
+	requestOrLimit string,
+) error {
+	for rc, containerRsc := range resourceSpec {
+		serverRsc, nonEmpty := minJobResources[rc]
+		if nonEmpty && containerRsc.Value() < serverRsc.Value() {
+			return fmt.Errorf("%s %s (%s) on container %v below server minimum (%s)", requestOrLimit, rc, &containerRsc, container_name, &serverRsc)
```

Comment:
```suggestion
			return fmt.Errorf("[validateContainerResource] containers must request at least %s of resource %s, but container %q only requests %s", serverRsc.Value(), rc, container_name, containerRsc.Value())
```

Reply:
I suggest updating the other error messages introduced by this PR similarly.

Reply:
Doesn't value return the amount of resource (e.g., 1000 if 1kb was requested)? Since these errors will be returned to users, I think we should make it as clear as possible what went wrong and what they should do to fix it. I think showing internal details of how we store limits etc. may confuse users, who shouldn't need to know those things.

Reply:
ah, passing &serverRsc will actually call the .String() function and will print 1Ki rather than 1000. I'm unconvinced printing .Value() is a good idea when we'll definitely have limits/reqs in the Gi, maybe even Ti

Reply:
Ah. In that case I agree. Passing `&serverRsc` would be much better.

Reply:
On review, I completely disagree with your format. That looks nothing like any of the errors raised by ValidatePodSpec or podspec.go

Reply:
To clarify, in all the errors we raise; none of them have an error type in square brackets a the start, and all of them respond in a very indicative format "<RESOURCE> X has done something" or "<RESOURCE> Y is something> .. this is why it's wrong". Your response is imperative/subjunctive in that it declares the ideal state of the world rather than starting with the current state of the world. I hope this makes sense